src/GraphControl/Control/Grapher.cpp

```diff
@@ -246,6 +255,26 @@ namespace GraphControl
                 co_await TryUpdateGraph(keepCurrentView);
             }
         }
+
+        int valid = 0;
+        int invalid = 0;
+        for (Equation ^ eq : Equations)
```

Comment:
If you move this telemetry into SetEquationErrors and SetEquationsAsValid you can re-use the loops in there to get a count and only loop over the equations once. You also avoid the risk of miscounting since you are looping over all Equations while those methods loop only the graphable equations.

Reply:
We want to count all equations even if they are hidden on the graph. So only logging the graphable equations could give us an unclear picture of how many equations are in the equation list.